### PR TITLE
refactor(theme-compiler): replace RegExp with SVGO for inline SVG optimization

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

Replace RegExp with SVGO for inline SVG optimization

Fixes https://jira.refinitiv.com/browse/ELF-2363

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/theme-compiler/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "Apache-2.0",
   "type": "module",
   "engines": {
-    "node": ">=18.0.0"
+    "node": ">=14.0.0"
```
Comment (Sakchai-Refinitiv) [inline packages/theme-compiler/package.json]: Why downgrade the minimum requirement?

File: packages/theme-compiler/src/less-plugin-inline-svg/encoders.js [hunk 2]
```diff
@@ -9,13 +9,11 @@ const encodeHtmlEntities = (input) => {
 };
 
 const optimizeSvg = (input) => {
-  // replace native RegExp with RE2 to prevent ReDos
-  // https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS
-  return input.replace(new RE2(/\s*\n\s*/g), '');
+  return optimize(input).data;
```
Comment (Sakchai-Refinitiv) [inline packages/theme-compiler/src/less-plugin-inline-svg/encoders.js:12]: I think we should include the `optimize` from svgo in  `wrapAndEncodeSvgToBase64` and remove this function

File: packages/theme-compiler/package.json (hunk 1 above)
Comment (wattachai-lseg) [reply packages/theme-compiler/package.json]: SVGO only requires node 14+.

File: packages/theme-compiler/src/less-plugin-inline-svg/encoders.js (hunk 2 above)
Comment (wattachai-lseg) [reply packages/theme-compiler/src/less-plugin-inline-svg/encoders.js:12]: `optimizeSvg` is being used in another location: check `warpSvg` function.

Comment (Sakchai-Refinitiv) [reply packages/theme-compiler/src/less-plugin-inline-svg/encoders.js:12]: OK, I see.

File: packages/theme-compiler/package.json (hunk 1 above)
Comment (Sakchai-Refinitiv) [reply packages/theme-compiler/package.json]: SVGO require 14+ and theme-compiler require 18+ it should work correctly and no effect to developers.

Comment (wattachai-lseg) [reply packages/theme-compiler/package.json]: Reverted back to 18 promoting usage of active LTS.

**Review by Sakchai-Refinitiv — DISMISSED** [review]
Good Job 👍

**Comment by Theeraphat-Sorasetsakul** [comment]
Verified by test various svg from the ticket.

**Comment by Theeraphat-Sorasetsakul** [comment]
Verified on ef-clock, ef-rating and input.

**Review by Theeraphat-Sorasetsakul — APPROVED** [review]

**Pull request merged**